src/OpenIddict.EF/OpenIddictContext.cs

```diff
@@ -10,12 +10,13 @@
 using OpenIddict.Models;
 
 namespace OpenIddict {
-    public class OpenIddictContext<TUser, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
+    public class OpenIddictContext<TUser, TApplication, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
         where TUser : IdentityUser<TKey>
+        where TApplication : Application
         where TRole : IdentityRole<TKey>
         where TKey : IEquatable<TKey> {
-        public DbSet<Application> Applications { get; set; }
+        public DbSet<TApplication> Applications { get; set; }
     }
 
-    public class OpenIddictContext<TUser> : OpenIddictContext<TUser, IdentityRole, string> where TUser : IdentityUser { }
+    public class OpenIddictContext<TUser> : OpenIddictContext<TUser, Application, IdentityRole, string> where TUser : IdentityUser { }
```

Comment:
FYI, it's rather unusual in ASP.NET Identity, where the User is always a custom type.